src/transformers/image_processing_utils_fast.py

```diff
@@ -131,3 +121,912 @@ def get_max_height_width(images: List["torch.Tensor"]) -> Tuple[int]:
+        """
+        validate_kwargs(captured_kwargs=kwargs.keys(), valid_processor_keys=self.valid_extra_kwargs)
+
+        do_resize = do_resize if do_resize is not None else self.do_resize
+        size = size if size is not None else self.size
+        default_to_square = kwargs.pop(
+            "default_to_square", self.default_to_square if self.default_to_square is not None else True
+        )
+        size = get_size_dict(size=size, default_to_square=default_to_square) if size is not None else None
+        resample = resample if resample is not None else self.resample
+        do_center_crop = do_center_crop if do_center_crop is not None else self.do_center_crop
+        crop_size = crop_size if crop_size is not None else self.crop_size
+        crop_size = get_size_dict(crop_size, param_name="crop_size") if crop_size is not None else None
+        do_rescale = do_rescale if do_rescale is not None else self.do_rescale
+        rescale_factor = rescale_factor if rescale_factor is not None else self.rescale_factor
+        do_normalize = do_normalize if do_normalize is not None else self.do_normalize
+        image_mean = image_mean if image_mean is not None else self.image_mean
+        image_std = image_std if image_std is not None else self.image_std
+        do_convert_rgb = do_convert_rgb if do_convert_rgb is not None else self.do_convert_rgb
+        device = kwargs.pop("device", None)
+
+        images, image_mean, image_std, size, crop_size, interpolation = self.prepare_process_arguments(
+            images=images,
+            do_resize=do_resize,
+            size=size,
+            resample=resample,
+            do_center_crop=do_center_crop,
+            crop_size=crop_size,
+            do_rescale=do_rescale,
+            rescale_factor=rescale_factor,
+            do_normalize=do_normalize,
+            image_mean=image_mean,
+            image_std=image_std,
+            do_convert_rgb=do_convert_rgb,
+            return_tensors=return_tensors,
+            data_format=data_format,
+            input_data_format=input_data_format,
+            device=device,
+            **kwargs,
+        )
+
+        # Group images by size for batched resizing
+        grouped_images, grouped_images_index = group_images_by_shape(images)
+        resized_images_grouped = {}
+        for shape, images in grouped_images.items():
+            stacked_images = torch.stack(images, dim=0)
+            if do_resize:
+                stacked_images = self.resize(image=stacked_images, size=size, interpolation=interpolation)
+            resized_images_grouped[shape] = stacked_images
+        resized_images = reorder_images(resized_images_grouped, grouped_images_index)
+
+        # Group images by size for further processing
+        # Needed in case do_resize is False, or resize returns images with different sizes
+        grouped_images, grouped_images_index = group_images_by_shape(resized_images)
```

Comment:
curious, did you get any runtime test for this, if grouping speeds up processing?

Reply:
Very cool, for large batched I think the speedup will come from triton kernels, as Olivier told me. Looks good for now as long as the grouping is decreasing latency for smaller batches :)